src/entities/character/init.rs

```diff
@@ -0,0 +1,183 @@
+use bevy::{
+    core_pipeline::clear_color::ClearColorConfig, prelude::*, render::view::RenderLayers,
+    utils::HashMap,
+};
+
+use crate::{
+    animation::{component::Animator, init::UpdateEntityAnimationEvent},
+    entities::worldmap::init::MapLoadedEvent,
+};
+
+#[derive(Component)]
+pub struct CharacterDrawing {
+    //need all information necessary to instansiate the other component
+    //TODO : place the handles of the TextureAtlas inside the character to unload them only when the character is not usefull anymore
+    pub current_action: CurrentAction,
+    animation_parameters: HashMap<CurrentAction, String>, //parameter to replace animation when needed
+}
+#[derive(Clone, Copy, PartialEq, Eq, Hash)]
+pub enum CurrentAction {
```

Comment:
```suggestion
pub enum CharacterAnimation {
```
At least, "Current" is not a good idea, because nothing is defined here

Reply:
Replace with CharacterAction because used in mouvement as well as animation